unittests/savanna_proposer_policy_tests.cpp

```diff
@@ -0,0 +1,159 @@
+   uint32_t expected_gap = 12 - (start_num % 12);   // 1. finishing the current round
+   expected_gap += 12;                              // 2. a full round
+   BOOST_REQUIRE_EQUAL(end_num, start_num + expected_gap);
+} FC_LOG_AND_RETHROW()
+
+
+// ---------------------------------------------------------------------------------------------------
+//     Proposer Policy change - check expected delay when policy change initiated on
+//                              the 6th block of a round
+// ---------------------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(policy_change_sixth_block_delay_check, savanna_cluster::cluster_t) try {
+   auto& A=_nodes[0];
+
+   auto start_num = A.head().block_num();
+   while(start_num % 12 != 6) {
+      A.produce_block();
+      start_num = A.head().block_num();
+   }
+
+   const vector<account_name> producers { "p0"_n, "p1"_n };
+   A.create_accounts(producers);
+   A.set_producers(producers);   // set new producers and produce blocks until the switch is pending
+   auto end_num = A.head().block_num();
+
+   // under Savanna, a new policy becomes active on the first block of a 12 block round after:
+   // 1. finishing the current round
+   // 2. a full round
+   // ----------------------------------------------------------------------------------------
+   uint32_t expected_gap = 12 - (start_num % 12);   // 1. finishing the current round
+   expected_gap += 12;                              // 2. a full round
+   BOOST_REQUIRE_EQUAL(end_num, start_num + expected_gap);
+} FC_LOG_AND_RETHROW()
+
+
+// ---------------------------------------------------------------------------------------------------
+//     Proposer Policy change - check expected delay when policy change initiated on
+//                              the last block of a round
+// ---------------------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE(policy_change_last_block_delay_check, savanna_cluster::cluster_t) try {
+   auto& A=_nodes[0];
+
+   auto start_num = A.head().block_num();
+   while(start_num % 12 != 11) {
+      A.produce_block();
+      start_num = A.head().block_num();
+   }
+
+   const vector<account_name> producers { "p0"_n, "p1"_n };
+   A.create_accounts(producers);
+   A.set_producers(producers);   // set new producers and produce blocks until the switch is pending
+   auto end_num = A.head().block_num();
+
+   // under Savanna, a new policy becomes active on the first block of a 12 block round after:
+   // 1. finishing the current round
+   // 2. a full round
+   // ----------------------------------------------------------------------------------------
+   uint32_t expected_gap = 12 - (start_num % 12);   // 1. finishing the current round
+   expected_gap += 12;                              // 2. a full round
+   BOOST_REQUIRE_EQUAL(end_num, start_num + expected_gap);
+} FC_LOG_AND_RETHROW()
```

Comment:
Where a block is within respect to a round (first, middle, or last block of the round) is determined by the slot number (timestamp) not the block number.

Additionally, even if you changes `head().block_num()` everywhere above to instead grab the slot of the timestamp of that head block, I'm still not sure if there isn't an off-by-one error in the three test cases above. I say this because you can check you get the desired remainder when dividing the slot number by 12, but the transaction that triggers the policy change goes into the next block. How is the first block of a round defined? Is it the one that has a slot which is a multiple of 12?

And you wouldn't really notice the off-by-one-error because what you may believe is on the last block of round N may be on the first block of round N+1. The `BOOST_REQUIRE_EQUAL` is designed to be robust enough that the test would still work fine for the same reasons the tests above currently pass even though you are incorrectly using the block number. But the comments would not be (and currently are not) accurate.